### PR TITLE
Split Prow image publishing job to distinguish non-prow images.

### PR DESCRIPTION
**Pull request by cjwagner**
The non-Prow images will still be built and published from test-infra, but the Prow images will eventually move with the source code. Note that `ghproxy` is being considered a Prow image due to tighter dependencies.

All images will still be published to gcr.io/k8s-prow/ for backwards compatibility.  @michelle192837 is working on an improvement to the generic-autobumper to allow for the non-prow images to eventually have different tags from the prow images once the prow images are published from kubernetes-sigs/prow.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -56,6 +59,27 @@ update-go-deps:
 .PHONY: verify-go-deps
 verify-go-deps:
 	hack/make-rules/verify/go-deps.sh
+# ======================== Image Building/Publishing ===========================
+# build and publish miscellaneous images that get pushed to gcr.io/k8s-prow
```
Comment (cjwagner) [inline Makefile:63]: These new make rules are closely based on the rules in `prow/Makefile`.

File: config/jobs/kubernetes/test-infra/test-infra-presubmits.yaml [hunk 2]
```diff
@@ -230,6 +230,39 @@ presubmits:
     annotations:
       testgrid-dashboards: presubmits-test-infra
       testgrid-tab-name: prow-image-build-test
+  # The following test builds the images specified in .k8s-prow-images.yaml
+  - name: pull-test-infra-misc-image-build-test
+    cluster: eks-prow-build-cluster
+    branches:
+    - master
+    # TODO: clean up this regexp once prow is publsihed from kubernetes-sigs/prow.
```
Comment (airbornepony) [inline config/jobs/kubernetes/test-infra/test-infra-presubmits.yaml:238]: published

(Also below.)

Comment (listx) [inline config/jobs/kubernetes/test-infra/test-infra-presubmits.yaml:238]: s/publsihed/published

**Review by airbornepony — CHANGES_REQUESTED** [review]

File: config/jobs/kubernetes/test-infra/test-infra-trusted.yaml [hunk 3]
```diff
@@ -259,6 +259,48 @@ postsubmits:
     rerun_auth_config:
       github_users:
       - alvaroaleman
+  - name: post-test-infra-push-misc-images
+    cluster: test-infra-trusted
+    # TODO: clean up this regexp once prow is publsihed from kubernetes-sigs/prow.
```
Comment (listx) [inline config/jobs/kubernetes/test-infra/test-infra-trusted.yaml:264]: s/publsihed/published

File: Makefile (hunk 1 above)
Comment (listx) [reply Makefile:63]: Nit: Turn line 63 into a regular sentence with a period at the end.

**Review by listx — COMMENTED** [review]
All nits.

/lgtm
/hold ?

**Comment by cjwagner** [comment]
Comments addressed, PTAL.

**Pull request merged**